src/lib/Microsoft.Health.Fhir.Ingest.Template/CalculatedFunctionContentTemplateExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using EnsureThat;
+
+namespace Microsoft.Health.Fhir.Ingest.Template
+{
+    public static class CalculatedFunctionContentTemplateExtensions
+    {
+        /// <summary>
+        /// Iterates over all expressions belonging to the content template. If the expression language has not been set it will be set
+        /// to default language of the content template.
+        /// </summary>
+        /// <param name="template">The template to retrieve expressions from</param>
+        public static void EnsureExpressionLanguageIsSet(this CalculatedFunctionContentTemplate template)
+        {
+            EnsureArg.IsNotNull(template, nameof(template));
+
+            foreach (var tuple in template.GetExpressions())
+            {
+                if (tuple.expression != null)
+                {
+                    tuple.expression.Language ??= template.DefaultExpressionLanguage;
```

Comment:
>tuple.expression.Language ??= template.DefaultExpressionLanguage; [](http://example.com/codeflow?start=20&length=65)

Is there a way to control this as part of the definition of TemplateExpression.  Ideally the default would be used automatically if not defined.

Reply:
Interesting question. Since it's optional for a user to specify the language it's left blank during deserialization. I briefly played around with NewtonSoft and the JsonConverter to see if I could set it but I didn't see an easy way to obtain the 'defaultExpressionLanguage' that is set on the root of the ContentTemplate itself. So I choose this extension as an alternative.

Reply:
Okay thanks for checking!